### ISSUE TITLE
[Bug]: v2 doesn't support RN <0.68 on iOS

### ISSUE THREAD

**Issue by micheleb** [braze-inc/braze-react-native-sdk#193]
### Which Platforms?

iOS

### Which React Native Version?

0.67.5

### Which @braze/react-native-sdk SDK version?

2.0.2

### Repro Rate

100% of the time

### Steps To Reproduce

Example:
  1. `yarn add @braze/react-native-sdk`
  2. Build the app on iOS.

### Expected Behavior

the app builds

### Actual Incorrect Behavior

the app doesn't build

### Additional Information

Hi there 👋🏻,

v2 is using the version of `RCTBundleURLProvider.jsBundleURLForBundleRoot` that doesn't take a `fallbackResource` as parameter, because the version that does take it has been deprecated.

However, that version [is only available from RN 0.68 onwards](https://github.com/facebook/react-native/commit/0912ee179c210fb6b2ed9afbb3f2fbc5fb8a2bb3#), making v2 not compatible with older versions of React Native.

I also had to explicitly import `<React/RCTBridgeDelegate.h>` from `BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.h` on [our fork of the library](https://github.com/braze-inc/braze-react-native-sdk/compare/master...justwatch:braze-react-native-sdk:master) to make it work (maybe because our `AppDelegate` doesn't extend the new `RCTAppDelegate`? Not sure, we also had to extend `UIResponder` directly, so it might be specific to us). With those 2 changes in place, our RN 0.67.5-based app builds fine.

Thanks!

**Comment by nromptea42** [comment]
Hey, here's a patch file if you're using patch-package

[@braze+react-native-sdk+2.0.2.patch](https://github.com/braze-inc/braze-react-native-sdk/files/10826244/%40braze%2Breact-native-sdk%2B2.0.2.patch)

```
diff --git a/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.h b/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.h
index 266e83a..ff1baa6 100644
--- a/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.h
+++ b/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.h
@@ -1,6 +1,7 @@
 #import <Foundation/Foundation.h>
 #import <React/RCTEventEmitter.h>
 #import <React/RCTBridgeModule.h>
+#import <React/RCTBridgeDelegate.h>

 #import <BrazeKit/BrazeKit-Swift.h>

diff --git a/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.m b/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.m
index b54d41f..88d77c7 100644
--- a/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.m
+++ b/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.m
@@ -808,7 +808,7 @@ - (BRZInAppMessageRaw *)getInAppMessageFromString:(NSString *)inAppMessageJSONSt
 #pragma mark - RCTBridgeDelegate protocol

 - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge {
-  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
+  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
 }
 `
 RCT_EXPORT_MODULE();
 ```

**Comment by hokstuff** [comment]
Hi @micheleb,

The Braze React Native SDK version 2.0.0 requires React Native 0.68 or higher. We will add a version matrix in our README to explicitly call this out in our next release.

Thanks!

**Comment by hokstuff** [comment]
Hi @micheleb,

We have released [version 3.0.0](https://github.com/braze-inc/braze-react-native-sdk/releases/tag/3.0.0) which adds in a version matrix with more information in our README. Thanks!
